### PR TITLE
@DoNotInherit typed.ActorRef

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala [hunk 1]
```diff
@@ -20,7 +21,10 @@ import akka.actor.typed.internal.InternalRecipientRef
  * messages are delivered to the [[DeadLetter]] channel of the
  * [[akka.event.EventStream]] on a best effort basis
  * (i.e. this delivery is not reliable).
+ *
+ * Not for user extension
```
Comment (hepin1989) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Internal use only.

Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: "Not for user extension" is the phrase we have used in other places. It's still public user api, but we don't support external subclasses

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala (hunk 1 above)
Comment (hepin1989) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Thanks for the clarification.

**Pull request merged**